Bump version to 2.4.5.99rc2

jmstover opened the pull request:
**Description of the Pull Request (PR):**

Bump the version for the RC release.


**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

jmstover committed:
Bump version to 2.4.6-RC2

jmstover committed:
rpmbuild doesn't like '-' in the version

jmstover committed:
Keep the 'rc#' in the version

GodloveD commented on configure.ac:
```diff
@@ -1,5 +1,5 @@
 AC_PREREQ(2.59)
-AC_INIT([singularity],[2.4.5],[gmkurtzer@gmail.com])
+AC_INIT([singularity],[2.4.6rc2],[gmkurtzer@gmail.com])
```
I think it should be `2.4.6-rc2` (<-note the dash).  Right?

jmstover replied:
rpmbuild blows up when there's a '-' in the version. Seems to work fine without it.

GodloveD replied:
Stupid rmpbuild.  😕 Guess that means that 2.5.0-rc probably isn't building right now either.  I guess it didn't get run through Travis since I just pushed directly to the repo and then tagged?

jmstover committed:
Stupid RPM versioning

GodloveD commented on configure.ac:
```diff
@@ -1,5 +1,5 @@
 AC_PREREQ(2.59)
-AC_INIT([singularity],[2.4.5],[gmkurtzer@gmail.com])
+AC_INIT([singularity],[2.4.6],[gmkurtzer@gmail.com])
```
I think this really needs to be 2.4.6rc2 because of all of the potential problems that others have noted.  I know that this may cause some challenges with release numbers, but at least folks will actually see what version they are running using the `--version` command.

jmstover replied:
Version for me is:
```
$ singularity --version
2.4.6-ver_fix.gcef599d
```

So the branch (ver_fix) is included as well as the git commit. If we have the 'rc' included, then a RPM update wont work because `2.4.6 < 2.4.6rc`.

For _testing_ it should be being built from the git tree and not a dist tarball anyway. It's not meant to stick around.

GodloveD replied:
> So the branch (ver_fix) is included as well as the git commit. 

Well, sure.  But a soon as we make the 2.4.6-rc2 tag, this thing will return...
```
$ singularity --version
2.5.6-dist
```
That's a problem because now a user thinks they have the real deal when they actually mistakenly installed a release candidate and we can't tell the difference if they send us a bug report either.  

As far as the update issue, if somebody has created and installed an RPM from a release candidate then presumably they are savvy enough to know how to remove the release candidate and install the release.  And if they forget or don't know how the `--version` command should be able to tell them that they don't have the latest release.  Right?

jmstover replied:
>  And if they forget or don't know how the --version command should be able to tell them that they don't have the latest release.

No. It will show:  `2.4.6-dist` or `2.5.0-dist` when building from a tarball (i.e. creating a package). When you do `make dist` the `.git/` directory isn't there, so it doesn't look up the git information and version becomes:  `@PACKAGE_VERSION@-dist` instead of `@PACKAGE_VERSION@-[branch](-[git commit])`

Look at `configure.ac` ~line 314

GodloveD commented on debian/changelog:
```diff
@@ -1,3 +1,8 @@
+singularity-container (2.4.6-1) unstable; urgency=high
```
I think this should actually be 2.4.6rc2-1 as well just so that we differentiate between an actual release here.  We can change it when we do the real release.  Thoughts?

jmstover replied:
Debian appears to be the same:

https://www.debian.org/doc/debian-policy/#version-numbers-based-on-dates

GodloveD reviewed (changes requested):
I still don't think this is just right.  meh.

jmstover committed:
Fix .spec file version

For the RPM spec, specify an explicit _rel of 0.#, where # is the
RC release. 0.2 for RC2, etc...

I don't know if we can subvert the Debian version to do the same.

jmstover committed:
Fall back to a close version of release and make sure RC# shows in version output

jmstover committed:
Remove the force setting of _rel
